torch/quantization/_equalize.py

```diff
@@ -0,0 +1,133 @@
+import torch
+import torch.nn as nn
+import copy
+
+def channel_range(input, axis=0):
+    ''' finds the range of weights associated with a specific channel
+
+    '''
+    size_of_tensor_dim = len(input.size())
+    mins = input
+    maxs = input
+
+    # reshape input to make specified axis the last axis in input
+    temp = list(range(size_of_tensor_dim))
+    temp[size_of_tensor_dim - 1] = axis
+    temp[axis] = size_of_tensor_dim - 1
+
+    mins = mins.permute(temp)
+    maxs = maxs.permute(temp)
+
+    # minimizing over all axises except the last axis
+    for i in range(size_of_tensor_dim - 1):
+        mins = torch.min(mins, 0)[0]
+        maxs = torch.max(maxs, 0)[0]
+
+    assert(mins.size()[0] == input.size()[axis])
+    return maxs - mins
+
+def scaling_channels(module1, module2, output_axis=0, input_axis=1):
+    ''' Given two adjacent tensors', the weights are scaled such that
+    the ranges of the first tensors' output channel are equal to the
+    ranges of the second tensors' input channel
+
+    '''
+    tensor1 = module1.weight
+    tensor2 = module2.weight
+    bias = module1.bias
+
+    output_channel_tensor1 = channel_range(tensor1, output_axis)
+    input_channel_tensor2 = channel_range(tensor2, input_axis)
+
+    # producing scaling factors to applied
+    input_channel_tensor2 += 1e-9
+    scaling_factors = torch.sqrt(output_channel_tensor1 / input_channel_tensor2)
+    r_scaling_factors = torch.reciprocal(scaling_factors)
+
+    bias = bias * r_scaling_factors
+
+    # formatting the scaling (1D) tensors to be applied on the given argument tensors
+    # pads axis to (1D) tensors to then be broadcasted
+    size1 = [1 for x in range(len(tensor1.size()))]
+    size1[output_axis] = tensor1.size()[output_axis]
```

Comment:
tensor1.size()[output_axis] --> tensor1.dim(output_axis)

Reply:
the dim() function for tensors just returns the number of axises, it doesn't query the axis you want like tensor1.size()[output_axis]

Reply:
Oh I think I meant, tensor1.size(output_axis) here